### PR TITLE
Rectangle patch angle attribute and patch __str__ improvements

### PR DESCRIPTION
**Pull request by cdeil**
We're in the process of writing an [astropy.regions](http://astropy-regions.readthedocs.io/en/latest/) package for astronomers, and we're adding an `as_patch` method to convert regions to matplotlib patch objects for plotting.

While writing a test for the rectangle region / patch, I noticed that `matplotlib.patches.Rectangle._angle` that was added in #1405 isn't exposed as a public attribute?

If that's the case, would it be possible to expose it via an `angle` attribute or `get_angle()` method?

I noticed that other parameters are exposed as a wild mix of attributes and `get_` methods:
```python
    def test_as_patch(self):
        patch = self.reg.as_patch()
        assert_allclose(patch.xy, (3, 4))
        assert_allclose(patch.get_width(), 4)
        assert_allclose(patch.get_height(), 3)
        assert_allclose(patch._angle, 5)
```

Presumably there are backward-compat constraints or name collisions, that make exposing patch data members in a uniform way difficult. Is there something that can be improved in general, or for `Rectangle.angle` specifically?

### CODE REVIEW

**Comment by cdeil** [comment]
@dopplershift - I've attached a commit with the change you suggest.

While looking at the file I noticed that `Rectangle.__str__` and `Ellipse.__str__` don't print the `angle`. They should, right?
Do you want me to fix this here in this PR also?

I see that you don't have unit tests with asserts on simple data members like `angle` or `__str__`.
Do you want me to add something, or do you prefer to only maintain tests for non-trivial things, to keep it lean?

**Comment by dopplershift** [comment]
Thanks! If you're willing to do the work, I'd say yes to both of your questions; I'm certainly not going to say no to more tests.

**Comment by cdeil** [comment]
OK, I'll do it here.

Side comment: when I try to run the tests following your docs, I get this:
```
$ python tests.py 
Python byte-compilation optimization level: 0
Traceback (most recent call last):
  File "tests.py", line 46, in <module>
    success = test(argv=sys.argv[0:1] + extra_args, switch_backend_warn=False)
TypeError: test() got an unexpected keyword argument 'argv'
```
I presume that's because it's picking up the `MPL 1.5` from my site-packages, because I didn't install the dev version?
That's pretty error prone.
I think I saw you're changing to pytest ... +10 to that and to get rid of such a `test.py` file that doesn't say which MPL version was picked up or what the problem was.

**Comment by cdeil** [comment]
@dopplershift - Please review.

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -1394,8 +1395,9 @@ class Ellipse(Patch):
     A scale-free ellipse.
     """
     def __str__(self):
-        return "Ellipse(%s,%s;%sx%s)" % (self.center[0], self.center[1],
-                                         self.width, self.height)
+        pars = self.center[0], self.center[1], self.width, self.height, self.angle
```
Comment (dopplershift) [inline lib/matplotlib/patches.py]: Can you wrap this line so it's less than 80 columns?

Comment (cdeil) [inline lib/matplotlib/patches.py]: I've amended the last commit to wrap to 80 char here.
And I've thrown in a wedge, arc and test docstring.
Hope you like it.

File: lib/matplotlib/tests/test_patches.py [hunk 2]
```diff
@@ -291,6 +291,29 @@ def test_wedge_range():
...
+    assert str(p) == 'Ellipse(xy=(1, 2), width=3, height=4, angle=5)'
+
+    p = mpatches.Rectangle(xy=(1, 2), width=3, height=4, angle=5)
+    assert str(p) == 'Rectangle(xy=(1, 2), width=3, height=4, angle=5)'
+
+    p = mpatches.Wedge(center=(1, 2), r=3, theta1=4, theta2=5, width=6)
+    assert str(p) == 'Wedge(center=(1, 2), r=3, theta1=4, theta2=5, width=6)'
+
+    p = mpatches.Arc(xy=(1, 2), width=3, height=4, angle=5, theta1=6, theta2=7)
+    assert str(p) == 'Arc(xy=(1, 2), width=3, height=4, angle=5, theta1=6, theta2=7)'
```
Comment (dopplershift) [inline lib/matplotlib/tests/test_patches.py]: Unfortunately, now this line is too long.

Comment (cdeil) [inline lib/matplotlib/tests/test_patches.py]: I've amended my commit to fix this PEP8 and checked PEP8 locally. Should be OK now.

**Comment by dopplershift** [comment]
Looks like just a freak windows failure (one job, unrelated). Merging...

Thanks!

**Pull request merged**